# This file is part of HiGFXback

# requires
REQUIRES="lite-build meson-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

if PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists libzvt-build; then
  LIBZVT=-Dsystem-libzvt=true
  REQUIRES="$REQUIRES libzvt-build"
else
  LIBZVT=--libexecdir=/dfb/lib
fi

# configure
PKG_CONFIG_PATH=/dfb/lib/pkgconfig meson setup $LIBZVT --prefix=/dfb build

# build
meson compile -C build

# install
install -d $DESTDIR/dfb/bin
install build/src/dfbterm $DESTDIR/dfb/bin
test -f build/src/dfbterm-pty-helper && install -d $DESTDIR/dfb/lib/dfbterm
test -f build/src/dfbterm-pty-helper && install build/src/dfbterm-pty-helper $DESTDIR/dfb/lib/dfbterm
test -f build/src/dfbterm-pty-helper && ln -sf dfbterm-pty-helper $DESTDIR/dfb/lib/dfbterm/gnome-pty-helper
install -d $DESTDIR/dfb/share/dfbterm
install -m 644 data/Misc-Fixed.dgiff $DESTDIR/dfb/share/dfbterm

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/dfbterm-build.pc << EOF
Name: DFBTerm
Version: 20260221
Description: DirectFB terminal emulator
Requires: $REQUIRES

exec=\\
/dfb/bin/dfbterm \\
EOF
test -f build/src/dfbterm-pty-helper && cat >> $DESTDIR/dfb/share/pkgconfig/dfbterm-build.pc << EOF
/dfb/lib/dfbterm/dfbterm-pty-helper \\
/dfb/lib/dfbterm/gnome-pty-helper \\
EOF
cat >> $DESTDIR/dfb/share/pkgconfig/dfbterm-build.pc << EOF
/dfb/share/dfbterm/Misc-Fixed.dgiff
EOF

exit

# patch
--- DFBTerm-20260221.orig/README
+++ DFBTerm-20260221/README
@@ -1,7 +1,7 @@
 DFBTerm
 =======
 
-DFBTerm is a terminal emulator runnning on DirectFB and based on libzvt.
+DFBTerm is a terminal emulator running on DirectFB and based on libzvt.
 
 libzvt (Zed's virtual terminal emulation library) is included directly in the source code,
 although it is possible to use the system's libzvt library.

# source
https://github.com/directfb2/DFBTerm/archive/616f40a4b846f65f14c0446c82ddfe2f582c081e/DFBTerm-20260221.tar.gz
